front-end/src/components/mealsList.jsx

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import { useState, useEffect } from 'react'
+
+import { baseUrl } from '../apiurl.js'
+import { MealCard } from './meal-card.jsx'
+
+function MealsList() {
+  const [meals, setMeals] = useState([])
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const response = await fetch(`${baseUrl()}meals`)
+        const data = await response.json()
+        setMeals(data)
+        setLoading(false)
+      } catch (error) {
+        console.error('Error fetching meals:', error)
```

Comment:
you should also disable the loading if some error occurs and or even better to show some message, right if the request fails the page gets stuck in the `loading` state.

to not repeat yourself, you can use `finally` block to set the loading to false, but still, I would use the `catch` block to display some sort of error message